From 190478af3367a970d39bae36f7c5842b0c9a5bbb Mon Sep 17 00:00:00 2001
From: Lionel Landwerlin <llandwerlin@gmail.com>
Date: Wed, 23 May 2012 18:33:40 +0100
Subject: [PATCH 9/9] gio: add missing libraries for tools

---
 gio/Makefile.am |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gio/Makefile.am b/gio/Makefile.am
index 20bcd81..b4b78b5 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -676,6 +676,8 @@ bin_PROGRAMS = gio-querymodules glib-compile-schemas glib-compile-resources gset
 
 glib_compile_resources_LDADD = \
 	$(top_builddir)/glib/libglib-2.0.la \
+	$(top_builddir)/gthread/libgthread-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
 	libgio-2.0.la
 
@@ -707,6 +709,8 @@ glib_compile_schemas_SOURCES = \
 gsettings_LDADD = \
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
 	libgio-2.0.la
 gsettings_SOURCES = gsettings-tool.c
 
@@ -720,6 +724,8 @@ bin_PROGRAMS += gdbus
 gdbus_SOURCES = gdbus-tool.c
 gdbus_LDADD = libgio-2.0.la \
 	$(top_builddir)/glib/libglib-2.0.la 		\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
 	$(top_builddir)/gobject/libgobject-2.0.la
 
 completiondir = $(datadir)/bash-completion/completions
@@ -737,6 +743,8 @@ gresource_SOURCES = gresource-tool.c
 gresource_CPPFLAGS = $(LIBELF_CFLAGS) $(AM_CPPFLAGS)
 gresource_LDADD = libgio-2.0.la \
 	$(top_builddir)/glib/libglib-2.0.la 		\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
 	$(top_builddir)/gobject/libgobject-2.0.la \
 	$(LIBELF_LIBS)
 
-- 
1.7.9.5

